index.js

```diff
@@ -2,268 +2,299 @@
+    .option('-i, --statsinterval <STATS_INTERVAL>', 'Downsample stats send to Insight Platform', 30)
+    .option('-j, --json', 'Stream logs in JSON format', false)
+    .option('-e, --eventstoken <EVENTS_TOKEN>', 'Specify log token for forwarding events', process.env.INSIGHT_EVENTSTOKEN)
+    .option('-l, --logstoken <LOGS_TOKEN>', 'Specify log token for logs', process.env.INSIGHT_LOGSTOKEN)
+    .option('-k, --statstoken <STATS_TOKEN>', 'Specify log token for forwarding statistics', process.env.INSIGHT_STATSTOKEN)
+    .option('-t, --token <TOKEN>', 'Specify token to use', process.env.INSIGHT_TOKEN)
+    .option('-v, --loggerlevel <LEVEL>', 'Define application logger level', process.env.INSIGHT_LOGGER_LEVEL)
+    .option('--matchByName <REGEX>', 'Forward logs for containers whose name matches <REGEX>')
+    .option('--matchByImage <REGEX>', 'Forward logs for containers whose image matches <REGEX>')
+    .option('--skipByName <REGEX>', 'Do not forward logs for containers whose name matches <REGEX>')
+    .option('--skipByImage <REGEX>', 'Do not forward logs for containers whose image matches <REGEX>')
+    .option('--no-events', 'Do not stream Docker events')
+    .option('--no-logs', 'Do not stream logs')
+    .option('--no-stats', 'Do not stream statistics')
+    .option('--no-secure', 'Send logs un-encrypted; no TSL/SSL')
+    .option('--port <PORT>', 'Specify port to forward logs to. Default depends on whether secure is set', undefined)
+    .option('--server <SERVER>', 'Specify server to forward logs to', '.data.logs.insight.rapid7.com')
+    .parse(process_args);
+
+  return program.opts();
 }
 
 function cli(process_args) {
-  const argv = minimist(process_args.slice(2), {
-    boolean: ['json', 'secure', 'stats', 'logs', 'dockerEvents', 'debug'],
-    string: ['token', 'region', 'logstoken', 'statstoken', 'eventstoken', 'server', 'port'],
-    alias: {
-      'token': 't',
-      'region': 'r',
-      'logstoken': 'l',
-      'newline': 'n',
-      'statstoken': 'k',
-      'eventstoken': 'e',
-      'secure': 's',
-      'json': 'j',
-      'statsinterval': 'i',
-      'add': 'a',
-      'debug': 'd'
-    },
-    default: {
-      json: false,
-      secure: true,
-      newline: true,
-      stats: true,
-      logs: true,
-      dockerEvents: true,
-      statsinterval: 30,
-      add: [ 'host=' + os.hostname() ],
-      debug: !!process.env.INSIGHT_DOCKER_DEBUG,
-      token: process.env.INSIGHT_TOKEN,
-      logstoken: process.env.INSIGHT_LOGSTOKEN || process.env.INSIGHT_TOKEN,
-      statstoken: process.env.INSIGHT_STATSTOKEN || process.env.INSIGHT_TOKEN,
-      eventstoken: process.env.INSIGHT_EVENTSTOKEN || process.env.INSIGHT_TOKEN,
-      server: '.data.logs.insight.rapid7.com',
-      port: undefined
-    }
-  });
+  let args = parse_args(process_args);
+
+  LOGGER = winston.createLogger({
+    //  If logger level isn't specified, we are silent
+    silent: !args.loggerlevel,
```

Comment:
so the default log level is to have no logs at all?

Reply:
yeah, I imagine customers starting this container don't want to find out that it itself streams a load of logs.
Default of silent seems normal to me

Reply:
afaik the container excludes logs from itself from being sent on, no?

Reply:
huh, you're right